From 083cc438ea5c7f286d4dff7a00699dc12acff9da Mon Sep 17 00:00:00 2001
From: Shuzhen Wang <shuzhenwang@google.com>
Date: Tue, 12 Nov 2024 13:47:06 -0800
Subject: [PATCH 258/436] Camera: Auto-generate NDK header

This fixes a missing ZOOM_METHOD enum definition in ndk.

Flag: EXEMPT python code
Test: Build
Bug: 265196098
Change-Id: Ib1af5bc00f2998df9e88d89b8f79cb7455439cd0
---
 .../include/camera/NdkCameraMetadataTags.h    | 22 +++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/camera/ndk/include/camera/NdkCameraMetadataTags.h b/camera/ndk/include/camera/NdkCameraMetadataTags.h
index b792de08ab..704b46efe5 100644
--- a/camera/ndk/include/camera/NdkCameraMetadataTags.h
+++ b/camera/ndk/include/camera/NdkCameraMetadataTags.h
@@ -9750,6 +9750,28 @@ typedef enum acamera_metadata_enum_acamera_control_low_light_boost_state {
 
 } acamera_metadata_enum_android_control_low_light_boost_state_t;
 
+// ACAMERA_CONTROL_ZOOM_METHOD
+typedef enum acamera_metadata_enum_acamera_control_zoom_method {
+    /**
+     * <p>The camera device automatically detects whether the application does zoom with
+     * ACAMERA_SCALER_CROP_REGION or ACAMERA_CONTROL_ZOOM_RATIO, and in turn decides which
+     * metadata tag reflects the effective zoom level.</p>
+     *
+     * @see ACAMERA_CONTROL_ZOOM_RATIO
+     * @see ACAMERA_SCALER_CROP_REGION
+     */
+    ACAMERA_CONTROL_ZOOM_METHOD_AUTO                                 = 0,
+
+    /**
+     * <p>The application intends to control zoom via ACAMERA_CONTROL_ZOOM_RATIO, and
+     * the effective zoom level is reflected by ACAMERA_CONTROL_ZOOM_RATIO in capture results.</p>
+     *
+     * @see ACAMERA_CONTROL_ZOOM_RATIO
+     */
+    ACAMERA_CONTROL_ZOOM_METHOD_ZOOM_RATIO                           = 1,
+
+} acamera_metadata_enum_android_control_zoom_method_t;
+
 // ACAMERA_CONTROL_AE_PRIORITY_MODE
 typedef enum acamera_metadata_enum_acamera_control_ae_priority_mode {
     /**
-- 
2.43.0

